src/klein/_plating.py

```diff
@@ -25,17 +34,74 @@ def _should_return_json(request):
     return bool(request.args.get(b"json"))
 
 
-def json_serialize(item):
+@inlineCallbacks
+def resolveDeferredObjects(root):
     """
-    A function similar to L{dumps}.
+    Wait on possibly nested L{Deferred}s that represent a JSON
+    serializable object.
+
+    @param root: A JSON-serializable object that may contain
+        L{Deferred}s, or a Deferred that will resolve to a
```

Comment:
Presumably Deferreds that resolve to...? (JSON-serializable? JSON-serializable with possibly Deferred? Probably needs words.